src/utils/src/arg_parser.rs

```diff
@@ -365,6 +367,32 @@ impl<'a> Arguments<'a> {
             return Ok(());
         }
 
+        // If `--describe-snapshot` is provided as a parameter, we artificially skip the parsing of
```

Comment:
I don't fully understand why any changes to the arg parser crate are needed. So far, we've managed to keep this crate quite standalone, without logic that is tied to a specific binary (firecracker, jailer, or the upcoming seccompiler).

As far as I understand, the only benefit of this modification is that the parsing of the arguments stops if it encounters the --describe-snapshot parameter.
I think the PR should still work as expected by leaving the arg parser unchanged, given the fact that Firecracker terminates after receiving this parameter.
If we however want to make sure that we stop parsing the rest of the arguments, maybe we could add a new parameter attribute (like `required()`), that stops parsing after is encountered. Maybe smth like `.stop_if_present()`, that could be used for such cases, maybe even for `--version` and `--help`.

The added value would be that we would keep the arg_parser crate more generic. WDYT?

Reply:
This is a very good observation @alindima! Indeed, it would be preferable to keep the arg parser implementation as generic as possible. At a closer look,  
we could get rid of all the specific logic inside the arg-parser for `help`, `version` and `describe-snapshot` and we would be having the same behavior (that is if we also insert the `help` parameter in the `ArgParser` just like we do for all the other ones). @luminitavoicu is there something else about this logic that we are missing? If not, I am ok with deleting it.
@alindima I do not have any strong opinion about adding a `stop_if_present` flag, does not seem to add significant value.

Reply:
I agree with you, @alindima and @dianpopa, the newly added `describe-snapshot` parameter should not make its way to the arg parser logic. I will move it.

I think it makes sense to also get rid of the extra logic for `help` and `version` inside the arg parser. Whether the parsing of the arguments stops when encountering one of these two parameters does not make a significant difference to me. I think adding a `stop_if_present` flag is not really necessary.

Reply:
sounds good to me